maven-plugin/src/test/java/io/github/belgif/rest/guide/validator/maven/plugin/AbstractValidatorMojoTest.java

```diff
@@ -29,4 +38,45 @@ void executeNoFile() {
         assertTrue(exception.getMessage().startsWith("File not found: "));
     }
 
+    @Test
+    void testMultiThreaded() throws InterruptedException, IOException {
+        ExecutorService executor = Executors.newFixedThreadPool(4);
+
+        List<File> outputFiles = new ArrayList<>();
+
+        List<Callable<Void>> tasks = getCallables(outputFiles);
+
+        executor.invokeAll(tasks);
+        executor.shutdown();
+        executor.awaitTermination(1, TimeUnit.MINUTES);
+
+        byte[] referenceBytes = this.getClass().getResource("report.json").openStream().readAllBytes();
+        // Assert results
+        for (File outputFile : outputFiles) {
+            assertTrue(outputFile.exists());
+            byte[] bytes = Files.readAllBytes(outputFile.toPath());
+            assertArrayEquals(referenceBytes, bytes);
```

Comment:
Make the test less fragile without byte-for-byte comparison: every time a little detail is changed (e.g. violation description), the report.json would need to be updated.

You could e.g. check that there were violations found and the output of all executions is equal (without comparing to a fixed output).

Reply:
Updated test